cms/djangoapps/contentstore/views/utilities/tasks.py

```diff
@@ -0,0 +1,97 @@
+from celery.task import task
+from django.conf import settings
+from django.core.mail import send_mail
+from django.core.urlresolvers import reverse
+from smtplib import SMTPException
+
+from edxmako.shortcuts import render_to_string
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.inheritance import own_metadata
+
+from opaque_keys.edx.keys import CourseKey
+from xblock.fields import Scope
+
+
+def _get_course_problems(course_key):
+    """
+    Retrieve all problems in the course
+    """
+    # Note: Some components may have been orphaned due to a bug in split
+    # modulestore (PLAT-799). We may be returning orphaned components in this
+    # list to be updated as well
+    return modulestore().get_items(
+        course_key,
+        qualifiers={"category": 'problem'},
+    )
+
+
+def _send_email_on_completion(course, user_username, user_email, modified_settings, success):
+    """
+    Send email to user on completion of celery task completion
+    """
+    context = {
+        'user_username': user_username,
+        'course_name': course.display_name,
+        'modified_settings': modified_settings,
+        'success': success
+    }
+
+    from_email = settings.DEFAULT_FROM_EMAIL
+    subject = render_to_string('emails/utilities_bulk_update_done_subject.txt', context)
+    subject = ''.join(subject.splitlines())
```

Comment:
Is this because the subject has newlines?

Reply:
Yes, or else you will get a `BadHeaderError`